dev/Repeater/APITests/ViewManagerTests.cs

```diff
@@ -759,6 +759,76 @@ public void ValidateFocusMoveOnElementClearedWithUniqueIds()
                     () => { ValidateCurrentFocus(repeater, 0 /*expectedIndex */, "3" /* expectedContent */); }
                 });
         }
+
+        [TestMethod]
+        // Why does this test work?
+        // When the elements get created from the RecyclingElementFactory, we get already "existing" data templates.
+        // However, the reason for the crash in #2384 is that those "empty" data templates actually still had their data context
+        // The data context of course only gets set after we give the ViewManager the data template.
+        // Thus, the DataContext we see in GetElementCore is the OLD datacontext.
+        // If that data context is not null, that means it did not get cleared when the element was recycled, which is the wrong behavior.
+        // To check if the clearing is working correctly, we log all data contexts, and check if all of them are null.
+        public void ValidateElementClearingClearsDataContext()
+        {
+            ItemsRepeater repeater = null;
+            MockRecyclingTestFactory elementFactory = null;
+            int elementClearingRaisedCount = 0;
+            ItemsRepeaterScrollHost scroller = null;
+            Log.Comment("Initialize ItemsRepeater");
+            RunOnUIThread.Execute(() =>
+            {
+                elementFactory = new MockRecyclingTestFactory();
+                elementFactory.RecyclePool = new RecyclePool();
+                elementFactory.Templates["Item"] = (DataTemplate)XamlReader.Load(
+                    @"<DataTemplate xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'> " + "<Button Content=\"{Binding}\" />" + @"</DataTemplate>");
+
+                repeater = CreateRepeater(Enumerable.Range(0, 100),
+                    elementFactory);
+
+                repeater.Layout = new StackLayout();
```

Comment:
This is the default, so don't need this line.

Reply:
Not if we use CreateRepeater which uses the MockVirtualizingLayout down the line.

Reply:
Good point :) This is okay then. Any layout should be fine as long as atleast one element is realized by the layout.